mahiru/definitions/connections.py

```diff
@@ -16,3 +16,73 @@ class ConnectionInfo:
     This is the response to a connection request.
     """
     conn_id: str
+
+
+class WireGuardEndpoint:
+    """Describes a WireGuard endpoint."""
+    def __init__(
+            self, address: str, port: int, key: str
+            ) -> None:
+        """Create a WireGuard endpoint description.
+
+        Args:
+            address: Public IP address of the endpoint.
+            port: Public port number of the endpoint.
+            key: Public key of the endpoint.
+        """
+        self.address = address
+        self.port = port
+        self.key = key
+
+    def endpoint(self) -> str:
+        """Returns the endpoint's address:port string."""
+        return f'{self.address}:{self.port}'
+
+
+class WireGuardConnectionRequest(ConnectionRequest):
+    """Describes a request to provide a connection to a data source.
+
+    There are always two sides to a connection, one which initiates
+    connection setup and one which responds. The initiator is the
+    client side of the REST request, and the container on that side
+    contains a compute asset which is the client side of the connection
+    within the VPN tunnel. The responder hosts the REST API for setting
+    up connections, and the container on that side contains a data
+    asset which is the server side of the connection within the VPN
+    tunnel.
```

Comment:
Is there any documentation missing that is specific to `WireGuardConnectionRequest`?

Reply:
It's in the comment for `__init__` I guess. The `net-admin-helper` docs have an explanation as well of how this works; maybe this should be part of the architecture documentation in the future.

Reply:
https://github.com/SecConNet/mahiru/issues/103 for that.